src/dynamic_programming/egg_dropping.rs

```diff
@@ -1,91 +1,84 @@
-/// # Egg Dropping Puzzle
+//! This module contains the `egg_drop` function, which determines the minimum number of egg droppings
+//! required to find the highest floor from which an egg can be dropped without breaking. It also includes
+//! tests for the function using various test cases, including edge cases.
 
-/// `egg_drop(eggs, floors)` returns the least number of egg droppings
-///     required to determine the highest floor from which an egg will not
-///     break upon dropping
+/// Returns the least number of egg droppings required to determine the highest floor from which an egg will not break upon dropping.
 ///
-/// Assumptions: n > 0
-pub fn egg_drop(eggs: u32, floors: u32) -> u32 {
-    assert!(eggs > 0);
-
-    // Explicity handle edge cases (optional)
-    if eggs == 1 || floors == 0 || floors == 1 {
-        return floors;
-    }
-
-    let eggs_index = eggs as usize;
-    let floors_index = floors as usize;
-
-    // Store solutions to subproblems in 2D Vec,
-    // where egg_drops[i][j] represents the solution to the egg dropping
-    // problem with i eggs and j floors
-    let mut egg_drops: Vec<Vec<u32>> = vec![vec![0; floors_index + 1]; eggs_index + 1];
-
-    // Assign solutions for egg_drop(n, 0) = 0, egg_drop(n, 1) = 1
-    for egg_drop in egg_drops.iter_mut().skip(1) {
-        egg_drop[0] = 0;
-        egg_drop[1] = 1;
-    }
-
-    // Assign solutions to egg_drop(1, k) = k
-    for j in 1..=floors_index {
-        egg_drops[1][j] = j as u32;
+/// # Arguments
+///
+/// * `eggs` - The number of eggs available.
+/// * `floors` - The number of floors in the building.
+///
+/// # Returns
+///
+/// * `Some(usize)` - The minimum number of drops required if the number of eggs is greater than 0.
+/// * `None` - If the number of eggs is less than or equal to 0.
```

Comment:
```suggestion
/// * `None` - If the number of eggs is 0.
```
`eggs` is of the type `usize`.

Reply:
Thanks.